### ISSUE TITLE
Close figure after saving

### ISSUE THREAD

**Issue by gshiba** [matplotlib/pytest-mpl#45]
I have a number of parameterized tests that generate ~150 pngs at ~500kB per image. The test suite not only uses a lot of memory (>10GB) but it also becomes incredibly slow after ~20 images. This seems to be because the figure objects are not closed.

The `matplotlib` warning can be seen with `--strict` (note I had to register `mpl_image_compare` as a marker in pytest.ini):

```
$ pytest test/test_plotting.py -s --strict -v
... 
...
...
test/test_plotting.py::test_plot[test/data/set20.csv] /home/venv/local/lib/python2.7/site-packages/matplotlib/pyplot.py:524: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).
  max_open_warning, RuntimeWarning)
PASSED
...
```

This fixed it: https://github.com/gshiba/pytest-mpl/commit/40a0b513d3870e82bb42e20bd2734826ed5a5d2e

**Comment by bsipocz** [comment]
@gshiba - Would you mind opening a PR with your fix?

**Comment by astrofrog** [comment]
Fixed in #46
